Tests fail

1. Clone repo
2. `mkdir build; cd build; cmake ..`
3. `cd ..; dbus-launch xvfb-run --auto-display cmake --build build --target check`

<details><summary>Output</summary>

```
[ 93%] Building CXX object tests/CMakeFiles/dbusmenuimportertest.dir/testutils.cpp.o
[ 96%] Linking CXX executable dbusmenuimportertest
[100%] Built target dbusmenuimportertest
UpdateCTestConfiguration  from :/home/foo/Desktop/libdbusmenu-lxqt/build/tests/DartConfiguration.tcl
UpdateCTestConfiguration  from :/home/foo/Desktop/libdbusmenu-lxqt/build/tests/DartConfiguration.tcl
Test project /home/foo/Desktop/libdbusmenu-lxqt/build/tests
Constructing a list of tests
Done constructing a list of tests
Updating test list for fixtures
Added 0 tests to meet fixture requirements
Checking test dependency graph...
Checking test dependency graph end
test 1
    Start 1: dbusmenuexportertest

1: Test command: /home/foo/Desktop/libdbusmenu-lxqt/build/tests/dbusmenuexportertest
1: Working Directory: /home/foo/Desktop/libdbusmenu-lxqt/build/tests
1: Test timeout computed to be: 10000000
1: ********* Start testing of DBusMenuExporterTest *********
1: Config: Using QtTest library 6.8.1, Qt 6.8.1 (x86_64-little_endian-lp64 shared (dynamic) release build; by GCC 14.2.1 20240910), arch unknown
1: PASS   : DBusMenuExporterTest::initTestCase()
1: PASS   : DBusMenuExporterTest::testGetSomeProperties(label only)
1: PASS   : DBusMenuExporterTest::testGetSomeProperties(disabled, label only)
1: PASS   : DBusMenuExporterTest::testGetSomeProperties(icon name)
1: PASS   : DBusMenuExporterTest::testGetAllProperties()
1: PASS   : DBusMenuExporterTest::testGetNonExistentProperty()
1: PASS   : DBusMenuExporterTest::testClickedEvent()
1: PASS   : DBusMenuExporterTest::testSubMenu()
1: PASS   : DBusMenuExporterTest::testDynamicSubMenu()
1: PASS   : DBusMenuExporterTest::testRadioItems()
1: PASS   : DBusMenuExporterTest::testNonExclusiveActionGroup()
1: PASS   : DBusMenuExporterTest::testClickDeletedAction()
1: PASS   : DBusMenuExporterTest::testDeleteExporterBeforeMenu()
1: PASS   : DBusMenuExporterTest::testUpdateAndDeleteSubMenu()
1: PASS   : DBusMenuExporterTest::testMenuShortcut()
1: PASS   : DBusMenuExporterTest::testGetGroupProperties()
1: PASS   : DBusMenuExporterTest::testActivateAction()
1: PASS   : DBusMenuExporterTest::testTrackActionsOnlyOnce()
1: PASS   : DBusMenuExporterTest::testHonorDontShowIconsInMenusAttribute()
1: PASS   : DBusMenuExporterTest::testDBusMenuObjectIsDeletedWhenExporterIsDeleted()
1: PASS   : DBusMenuExporterTest::testSeparatorCollapsing(one-separator)
1: PASS   : DBusMenuExporterTest::testSeparatorCollapsing(two-separators)
1: PASS   : DBusMenuExporterTest::testSeparatorCollapsing(middle-separators)
1: PASS   : DBusMenuExporterTest::testSeparatorCollapsing(separators-at-begin)
1: PASS   : DBusMenuExporterTest::testSeparatorCollapsing(separators-at-end)
1: PASS   : DBusMenuExporterTest::testSeparatorCollapsing(separators-everywhere)
1: PASS   : DBusMenuExporterTest::testSeparatorCollapsing(empty-menu)
1: PASS   : DBusMenuExporterTest::testSeparatorCollapsing(separators-only)
1: PASS   : DBusMenuExporterTest::testSetStatus()
1: PASS   : DBusMenuExporterTest::testGetIconDataProperty()
1: PASS   : DBusMenuExporterTest::cleanupTestCase()
1: Totals: 31 passed, 0 failed, 0 skipped, 0 blacklisted, 10668ms
1: ********* Finished testing of DBusMenuExporterTest *********
1/3 Test #1: dbusmenuexportertest .............   Passed   10.70 sec
test 2
    Start 2: dbusmenuimportertest

2: Test command: /home/foo/Desktop/libdbusmenu-lxqt/build/tests/dbusmenuimportertest
2: Working Directory: /home/foo/Desktop/libdbusmenu-lxqt/build/tests
2: Test timeout computed to be: 10000000
2: ********* Start testing of DBusMenuImporterTest *********
2: Config: Using QtTest library 6.8.1, Qt 6.8.1 (x86_64-little_endian-lp64 shared (dynamic) release build; by GCC 14.2.1 20240910), arch unknown
2: PASS   : DBusMenuImporterTest::initTestCase()
2: PASS   : DBusMenuImporterTest::testStandardItem()
2: FAIL!  : DBusMenuImporterTest::testAddingNewItem() Compared values are not the same
2:    Actual   (outputMenu->actions().count()): 1
2:    Expected (inputMenu.actions().count())  : 2
2:    Loc: [/home/foo/Desktop/libdbusmenu-lxqt/tests/dbusmenuimportertest.cpp(86)]
2: PASS   : DBusMenuImporterTest::testShortcut()
2: QWARN  : DBusMenuImporterTest::testDeletingImporterWhileWaitingForAboutToShow() ?[31mvoid DBusMenuImporter::slotGetLayoutFinished(QDBusPendingCallWatcher*)?[0m: "No such object path '/TestMenuBar'"
2: QWARN  : DBusMenuImporterTest::testDeletingImporterWhileWaitingForAboutToShow() ?[31mvoid DBusMenuImporter::slotAboutToShowDBusCallFinished(QDBusPendingCallWatcher*)?[0m: Call to AboutToShow() failed: "No such object path '/TestMenuBar'"
2: QWARN  : DBusMenuImporterTest::testDeletingImporterWhileWaitingForAboutToShow() ?[31mbool DBusMenuImporterPrivate::waitForWatcher(QDBusPendingCallWatcher*, int)?[0m: "No such object path '/TestMenuBar'"
2: QWARN  : DBusMenuImporterTest::testDeletingImporterWhileWaitingForAboutToShow() ?[31mvoid DBusMenuImporter::slotMenuAboutToShow()?[0m: Application did not answer to AboutToShow() before timeout
2: PASS   : DBusMenuImporterTest::testDeletingImporterWhileWaitingForAboutToShow()
2: PASS   : DBusMenuImporterTest::testDynamicMenu()
2: PASS   : DBusMenuImporterTest::testActionActivationRequested()
2: PASS   : DBusMenuImporterTest::testActionsAreDeletedWhenImporterIs()
2: QWARN  : DBusMenuImporterTest::testIconData() QFSFileEngine::open: No file name specified
2: QWARN  : DBusMenuImporterTest::testIconData() QFile::remove: Empty or null file name
2: QWARN  : DBusMenuImporterTest::testIconData() QFSFileEngine::open: No file name specified
2: QWARN  : DBusMenuImporterTest::testIconData() QFile::remove: Empty or null file name
2: PASS   : DBusMenuImporterTest::testIconData()
2: PASS   : DBusMenuImporterTest::testInvisibleItem()
2: QDEBUG : DBusMenuImporterTest::testDisabledItem() ?[34mvoid DBusMenuImporterTest::testDisabledItem()?[0m: Disabling
2: PASS   : DBusMenuImporterTest::testDisabledItem()
2: PASS   : DBusMenuImporterTest::cleanupTestCase()
2: Totals: 11 passed, 1 failed, 0 skipped, 0 blacklisted, 10076ms
2: ********* Finished testing of DBusMenuImporterTest *********
2/3 Test #2: dbusmenuimportertest .............***Failed   10.12 sec
test 3
    Start 3: dbusmenushortcuttest

3: Test command: /home/foo/Desktop/libdbusmenu-lxqt/build/tests/dbusmenushortcuttest
3: Working Directory: /home/foo/Desktop/libdbusmenu-lxqt/build/tests
3: Test timeout computed to be: 10000000
3: ********* Start testing of DBusMenuShortcutTest *********
3: Config: Using QtTest library 6.8.1, Qt 6.8.1 (x86_64-little_endian-lp64 shared (dynamic) release build; by GCC 14.2.1 20240910), arch unknown
3: PASS   : DBusMenuShortcutTest::initTestCase()
3: PASS   : DBusMenuShortcutTest::testConverter((Qt::ALT | Qt::Key_F4))
3: PASS   : DBusMenuShortcutTest::testConverter((Qt::CTRL | Qt::Key_S))
3: PASS   : DBusMenuShortcutTest::testConverter((Qt::CTRL | Qt::Key_X, Qt::ALT | Qt::SHIFT | Qt::Key_Q))
3: PASS   : DBusMenuShortcutTest::testConverter((Qt::META | Qt::Key_E))
3: PASS   : DBusMenuShortcutTest::testConverter((Qt::CTRL | Qt::Key_Plus))
3: PASS   : DBusMenuShortcutTest::testConverter((Qt::CTRL | Qt::Key_Minus))
3: PASS   : DBusMenuShortcutTest::cleanupTestCase()
3: Totals: 8 passed, 0 failed, 0 skipped, 0 blacklisted, 0ms
3: ********* Finished testing of DBusMenuShortcutTest *********
3/3 Test #3: dbusmenushortcuttest .............   Passed    0.04 sec

67% tests passed, 1 tests failed out of 3

Total Test time (real) =  20.86 sec

The following tests FAILED:
          2 - dbusmenuimportertest (Failed)
Errors while running CTest
Output from these tests are in: /home/foo/Desktop/libdbusmenu-lxqt/build/tests/Testing/Temporary/LastTest.log
Use "--rerun-failed --output-on-failure" to re-run the failed cases verbosely.
make[3]: *** [tests/CMakeFiles/check.dir/build.make:72: tests/CMakeFiles/check] Error 8
make[2]: *** [CMakeFiles/Makefile2:400: tests/CMakeFiles/check.dir/all] Error 2
make[1]: *** [CMakeFiles/Makefile2:407: tests/CMakeFiles/check.dir/rule] Error 2
make: *** [Makefile:260: check] Error 2
```
</details>

--- a/tests/dbusmenuexportertest.cpp
+++ b/tests/dbusmenuexportertest.cpp
@@ -392,10 +392,10 @@ void DBusMenuExporterTest::testNonExclusiveActionGroup()
     // Check items are checkmark, not radio
     item = list.takeFirst();
     QCOMPARE(item.properties.value("toggle-type").toString(), QString("checkmark"));
-    int a1Id = item.id;
+    //int a1Id = item.id;
     item = list.takeFirst();
     QCOMPARE(item.properties.value("toggle-type").toString(), QString("checkmark"));
-    int a2Id = item.id;
+    //int a2Id = item.id;
 }
 
 void DBusMenuExporterTest::testClickDeletedAction()
@@ -459,7 +459,7 @@ void DBusMenuExporterTest::testMenuShortcut()
     // Create a menu containing an action with a shortcut
     QMenu inputMenu;
     QVERIFY(QDBusConnection::sessionBus().registerService(TEST_SERVICE));
-    DBusMenuExporter *exporter = new DBusMenuExporter(TEST_OBJECT_PATH, &inputMenu);
+    DBusMenuExporter *exporter = new DBusMenuExporter(TEST_OBJECT_PATH, &inputMenu); Q_UNUSED(exporter)
 
     QAction *a1 = inputMenu.addAction("a1");
     a1->setShortcut(Qt::CTRL | Qt::Key_A);
@@ -501,10 +501,10 @@ void DBusMenuExporterTest::testGetGroupProperties()
     // Create a menu containing two actions
     QMenu inputMenu;
     QVERIFY(QDBusConnection::sessionBus().registerService(TEST_SERVICE));
-    DBusMenuExporter *exporter = new DBusMenuExporter(TEST_OBJECT_PATH, &inputMenu);
+    DBusMenuExporter *exporter = new DBusMenuExporter(TEST_OBJECT_PATH, &inputMenu); Q_UNUSED(exporter)
 
-    QAction *a1 = inputMenu.addAction("a1");
-    QAction *a2 = inputMenu.addAction("a2");
+    inputMenu.addAction("a1");
+    inputMenu.addAction("a2");
 
     // Check exporter is on DBus
     QDBusInterface iface(TEST_SERVICE, TEST_OBJECT_PATH);
@@ -589,7 +589,7 @@ void DBusMenuExporterTest::testTrackActionsOnlyOnce()
     // submenu should not have more than one DBusMenu child object.
     QMenu mainMenu;
     QVERIFY(QDBusConnection::sessionBus().registerService(TEST_SERVICE));
-    DBusMenuExporter *exporter = new DBusMenuExporter(TEST_OBJECT_PATH, &mainMenu);
+    DBusMenuExporter *exporter = new DBusMenuExporter(TEST_OBJECT_PATH, &mainMenu); Q_UNUSED(exporter)
 
     QMenu* subMenu = new QMenu("File");
     subMenu->addAction("a1");
@@ -650,7 +650,7 @@ void DBusMenuExporterTest::testDBusMenuObjectIsDeletedWhenExporterIsDeleted()
     QVERIFY(QDBusConnection::sessionBus().registerService(TEST_SERVICE));
     DBusMenuExporter *exporter = new DBusMenuExporter(TEST_OBJECT_PATH, &inputMenu);
 
-    QAction *a1 = inputMenu.addAction("a1");
+    inputMenu.addAction("a1");
     QVERIFY2(hasInternalDBusMenuObject(&inputMenu), "Test setup failed");
     delete exporter;
     QVERIFY(!hasInternalDBusMenuObject(&inputMenu));
@@ -680,7 +680,7 @@ void DBusMenuExporterTest::testSeparatorCollapsing()
     QMenu inputMenu;
 
     QVERIFY(QDBusConnection::sessionBus().registerService(TEST_SERVICE));
-    DBusMenuExporter *exporter = new DBusMenuExporter(TEST_OBJECT_PATH, &inputMenu);
+    DBusMenuExporter *exporter = new DBusMenuExporter(TEST_OBJECT_PATH, &inputMenu); Q_UNUSED(exporter)
 
     if (input.isEmpty()) {
         // Pretend there was an action so that doEmitLayoutUpdated() is called

--- a/tests/dbusmenuimportertest.cpp
+++ b/tests/dbusmenuimportertest.cpp
@@ -58,7 +58,7 @@ void DBusMenuImporterTest::cleanup()
 void DBusMenuImporterTest::testStandardItem()
 {
     QMenu inputMenu;
-    QAction *action = inputMenu.addAction("Test");
+    inputMenu.addAction("Test");
     DBusMenuExporter exporter(TEST_OBJECT_PATH, &inputMenu);
 
     DBusMenuImporter importer(TEST_SERVICE, TEST_OBJECT_PATH);
@@ -73,7 +73,7 @@ void DBusMenuImporterTest::testStandardItem()
 void DBusMenuImporterTest::testAddingNewItem()
 {
     QMenu inputMenu;
-    QAction *action = inputMenu.addAction("Test");
+    inputMenu.addAction("Test");
     DBusMenuExporter exporter(TEST_OBJECT_PATH, &inputMenu);
 
     DBusMenuImporter importer(TEST_SERVICE, TEST_OBJECT_PATH);
@@ -82,7 +82,8 @@ void DBusMenuImporterTest::testAddingNewItem()
     QCOMPARE(outputMenu->actions().count(), inputMenu.actions().count());
 
     inputMenu.addAction("Test2");
-    QTest::qWait(500);
+    inputMenu.addAction("Test3");
+    QTest::qWait(2000); // see "LAYOUT_UPDATE_TIMEOUT" in the c-tor of "DBusMenuImporter"
     QCOMPARE(outputMenu->actions().count(), inputMenu.actions().count());
 }
 

